File: 1047/1047.py
d= input()
z = d.split()
hi = int(z[0])
mi = int(z[1])
hf = int(z[2])
mf = int(z[3])

if(hi == mi == hf == mf):
    print("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)")
elif(hi > hf):
    calci=(hi*60)+mi
    calcf=((hf+24)*60)+mf 
    result =calci - calcf 
    time = abs(result)//60
    minutos =  abs(result) % 60
    print(f"O JOGO DUROU {abs(time)} HORA(S) E {abs(minutos)} MINUTO(S)")
elif(hi == hf  and mf < mi):
    calci=(hi*60)+mi
    calcf=((hf+24)*60)+mf 
    result =calci - calcf 
    time = abs(result)//60
    minutos =  abs(result) % 60
    print(f"O JOGO DUROU {abs(time)} HORA(S) E {abs(minutos)} MINUTO(S)")
else:
    calci=(hi*60)+mi
    calcf=(hf*60)+mf 
    result =calci - calcf 
    time = abs(result)//60
    minutos = abs(result) % 60
    print(f"O JOGO DUROU {abs(time)} HORA(S) E {abs(minutos)} MINUTO(S)")

File: 1047/1047.c
#include <stdio.h>
#include <stdlib.h>

int main() {
    int hi, mi, hf, mf;
    
    scanf("%d %d %d %d", &hi, &mi, &hf, &mf);
    
    if (hi == mi && hf == mf) {
        printf("O JOGO DUROU 24 HORA(S) E 0 MINUTO(S)\n");
    } else if (hi > hf) {
        int calci = hi * 60 + mi;
        int calcf = (hf + 24) * 60 + mf;
        int result = calci - calcf;
        int time = abs(result) / 60;
        int minutos = abs(result) % 60;
        printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", abs(time), abs(minutos));
    } else if (hi == hf && mf < mi) {
        int calci = hi * 60 + mi;
        int calcf = (hf + 24) * 60 + mf;
        int result = calci - calcf;
        int time = abs(result) / 60;
        int minutos = abs(result) % 60;
        printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", abs(time), abs(minutos));
    } else {
        int calci = hi * 60 + mi;
        int calcf = hf * 60 + mf;
        int result = calci - calcf;
        int time = abs(result) / 60;
        int minutos = abs(result) % 60;
        printf("O JOGO DUROU %d HORA(S) E %d MINUTO(S)\n", abs(time), abs(minutos));
    }
    
    return 0;
}